### DISCUSSION TITLE
Install on piCore

### DISCUSSION THREAD

**Discussion by michnixweiss** [MiczFlor/RPi-Jukebox-RFID#1750] [Q&A]
I am using a RPI Zero W with onoffshim, a wm8960 sound Hat, the RDM6300, rotary and some buttons. 
Cause of the hardware limitations and so caused lagging and slow boot of Raspbian (DietPi was not much faster for me) I want to give piCore a try.
My first steps are done
- Setup headless with wifi and ssh working
- Setup hardware (drivers, scripts, config, etc)
- Build needed extensions where they don't exist (raspi-gpio, python3.8-evdev, etc)
So now its time to install phoniebox.
Since there is no apt I cannot use the installer, but want to try the future3 build.
Can someone give me a list of needed packages and a step by step install guide? I could go through the install script step by step but might miss something.
The package list should include the version if necessary.

Many thanks

**Comment by pabera** [comment]
Hey. If you don't have apt available, I am not sure if you can get this to run, even `future3`. To some extend, it does have quite some dependencies, both on apt but also on [Python](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/develop/requirements.txt) and some custom build libs. Compiling a full list is difficult, but a good start would be the the [installation](https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/future3/develop/installation) section.

What is piCore running on if not Debian?

**Comment by michnixweiss** [comment]
Hey @pabera,

Thanks for the reply.
piCore is the RPi fork of TinyCoreLinux.

ATM I am unsure about using future3. The reason of switching to piCore is the long boottime (I couldn't get it under about 50sec) and the laggi performance (caused by the reduced hardware specs of my used RPi Zero W) of Raspbian/RaspberryPiOS.

Switching completely to python in future3 makes it more performant, but to have to use PulseAudio kills all benefits for me.

So quiet unsure if I first build packages for 2.3 and give it a try...

**Reply by pabera** [reply]
The boot time is indeed a problem which we haven't been able to action on much. `future3` is currently even slower than 2.x.
The thing is, we are relying on Python for the business logic, but playing music, RFID or sound is done by 3rd party software (e.g. `mpd`)

> but to have to use PulseAudio kills all benefits for me.

What's the issue with PulseAudio? Is it because it wouldn't run on your system or it reduces performance?

**Reply by michnixweiss** [reply]
I read the discussion about switching to PulseAudio.
I agree with the benefits eg bluetooth (which Im not using) but also think that it and its dependencies are heavy load for low hardware (like the Zero Im using).
What are the basic requirements to the player? Maybe there is a lightwight alternativ to mpd!?

**Reply by pabera** [reply]
You are touching a sensitive point. In fact, the current branch of 3.x is still dependent on mpd. To integrate Spotify, we are trying to abstract the player to support different backends where mpd and Spotify can be used, but where other alternatives could be added. Once that is implemented, mpd is not a strong dependency anymore and less consuming alternatives could be implemented.

**Comment by MiczFlor** [comment]
Hi @michnixweiss 
you are asking why `mpd` and what features are needed if there was a more lightweight alternative.
Going back into the Phoniebox archives: `mpd` replaces `cvlc` -- the command line version of `vlc` -- in the past. As a more lightweight option. If you are interested in the original `vlc` version, the last release is here:
https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/Version-0.9.7-with-VLC-(discontinued)
The reason I bring `vlc` back into the dicsussion is not because it might be a good alternative, but because back in the days, the dicussion about features also came up. And it turned out that the most basic features needed where:

1. playing audio files like mp3 / ogg / other formats
2. supporting playlists

But looking at the Phoniebox feature set, other things became important, like (I linked each item to the matching code)

3. [play one file from list then stop, remember where you stopped and the next time somebody swipes the card: play the next item on the list and then stop](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/5084ace1f1077f04d4fae4640545e7df9b97f9d7/scripts/playout_controls.sh#L975)
4. [add to playlist and play. This command clears the playlist, loads a new playlist and plays it](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/5084ace1f1077f04d4fae4640545e7df9b97f9d7/scripts/playout_controls.sh#L891)
5. [seek = move xvy seconds forward or jump to a position in a track](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/5084ace1f1077f04d4fae4640545e7df9b97f9d7/scripts/playout_controls.sh#L814)

There are more features in this script. Replacing the player would require this to be rewritten. Some of the features are covered by `mpd` some are workarounds (like: save current position on stop to resume from this position when starting again -- very good for audiobooks...).
What I am trying to say: if you are playing around with the idea of replacing the audio player for something more simple, take a look at the file and imagine what that would mean for the new player. It's not rocket science. As long as the player supports single files and playlists, the main issue is: does it also return information when you stop it, e.g. what track number in the playlist, what time position in the current track.
Hope this makes sense, keep us posted!
micz

**Comment by michnixweiss** [comment]
Since all my attempts have failed to achieve the desired result, I am closing the topic of piCore for myself.
